### PR TITLE
Do not compute PackageDef members in extractor

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/tasty/TastyImpl.scala [hunk 1]
```diff
@@ -194,17 +194,19 @@ object TastyImpl extends scala.tasty.Tasty {
...
-    def unapply(x: PackageDef)(implicit ctx: Context): Option[(String, List[Statement])] = x match {
+    def unapply(x: PackageDef)(implicit ctx: Context): Option[(String, PackageDef)] = x match {
       case x: PackageDefinition =>
-        // FIXME Do not do this eagerly as it forces everithing in the package to be loaded.
-        //       An alternative would be to add it as an extension method instead.
-        val definitions =
-          if (x.symbol.is(core.Flags.JavaDefined)) Nil // FIXME should also support java packages
-          else x.symbol.info.decls.iterator.map(FromSymbol.definition).toList
-        Some(x.symbol.name.toString, definitions)
+        Some((x.symbol.name.toString, FromSymbol.packageDef(x.symbol.owner)))
```
Comment (Blaisorblade) [inline compiler/src/dotty/tools/dotc/tasty/TastyImpl.scala]: Pretty hard to guess here: the returned PackageDef is the owner. Lacking constructors that's hard to document.

**Comment by nicolasstucki** [comment]
Rebased

**Comment by nicolasstucki** [comment]
The confusion here is that in tasty reflet we call the PackageDef tree PackageClause. The PackageDef in tasty reflect represents the symbol of the package.

**Comment by nicolasstucki** [comment]
@Blaisorblade any other comment or can we merge this one?

**Comment by Blaisorblade** [comment]
Merging.

> The PackageDef in tasty reflect represents the symbol of the package.

Now I see what you mean, but I don't understand what the distinction is supposed to represent conceptually.

As far as I can tell, `tasty.PackageDef` has no corresponding node in tasty, and it's just used as a way to proxy package clauses without loading the whole package.

**Pull request merged**